transformer_engine/paddle/profile.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+"""Utils for profiling"""
+
+import ctypes
+from contextlib import contextmanager
+import os
+
+_cuda_home = os.environ.get('CUDA_HOME', '/usr/local/cuda')
+
+_cudart = ctypes.CDLL(os.path.join(_cuda_home, 'lib64/libcudart.so'))
+
+
+def cuda_profile_start():
+    """Calls cudaProfilerStart"""
+    _cudart.cudaProfilerStart()
+
+
+def cuda_profile_stop():
+    """Calls cudaProfilerStop"""
+    _cudart.cudaProfilerStop()
+
+
+_nvtx = ctypes.CDLL(os.path.join(_cuda_home, 'lib64/libnvToolsExt.so'))
```

Comment:
@timmoon10 , @Tom-Zheng 
Could we use https://pypi.org/project/nvtx/ ?

Reply:
If yes, nvtx need to be added in [extension-pkg-whitelist](https://github.com/NVIDIA/TransformerEngine/blob/main/qa/L0_paddle_lint/pylintrc#L2) and [install_reqs](https://github.com/NVIDIA/TransformerEngine/blob/main/setup.py#L306).

Reply:
https://pypi.org/project/nvtx/ is also developed by NVIDIA, so I think we don't need to the reinvent wheel.

Reply:
Using `paddle.fluid.core.nvprof_nvtx_push` and `paddle.fluid.core.nvprof_nvtx_pop` instead.